unit_tests/stencil_composition/structured_grids/test_tmp_storage_sid_cuda.cu

```diff
@@ -0,0 +1,30 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "test_tmp_storage_sid_cuda.cpp"
+
+#include "../../test_helper.hpp"
+#include <gridtools/stencil_composition/backend_cuda/simple_device_memory_allocator.hpp>
+
+namespace gridtools {
+    namespace {
+        using tmp_cuda_storage_sid_float = tmp_cuda_storage_sid<float_type, simple_device_memory_allocator>;
+        TEST_F(tmp_cuda_storage_sid_float, maker_with_device_allocator) {
+            auto testee = get_tmp_storage();
```

Comment:
This is confusing.

You use test fixture just to get the type of the returned sid.
Can you do it explicitly :
```
  using testee_t = decltype(make_tmp_storage_cuda<int>(
                    tmp_cuda::blocksize<1, 1>{},
                    extent<>{},
                    0,
                    0,
                    0,
                    std::declval<simple_device_memory_allocator&>()));
```
?
And move `static_asserts` out of the test function? 

If you include `*.cpp` file, please include it the last line of the file.

Reply:
Do you think I should include the .cpp test in this case?

Reply:
may be. it verifies nvcc compilation